src/components/Input/Input.stories.tsx

```diff
@@ -283,8 +283,10 @@ export const WithCustomLogic = (): ReactElement => {
         }}
         isDisabled={inherited}
         value={value}
-        onChange={(_, { value: val }) => {
-          setValue(val)
+        onChange={(_, val: string | Record<string, unknown>) => {
+          if (typeof val === 'object') {
+            setValue(String(val.value))
```

Comment:
why is it `unknown` if then you convert it to a String?

Reply:
This is an issue with typescript: the problem is the value parameter can be a string or an object, so our linter complains that "val.value" cannot be accessed in case "value" is a string.
It seems quite difficult to manage in typescript, so in the story the switch with the "typeof" cases is necessary, but it wouldn't if the project is in plain javascript, or even if you admit the "val" parameter to be of type "any" in the "onChange"  typing (our linter complains with using "any" in every context)